hack/lib/util.sh

```diff
@@ -649,7 +649,9 @@ EOF
 # Determines if docker can be run, failures may simply require that the user be added to the docker group.
 function kube::util::ensure_docker_daemon_connectivity {
   IFS=" " read -ra DOCKER <<< "${DOCKER_OPTS}"
-  DOCKER=(docker "${DOCKER[@]}")
+  # Expand ${DOCKER[@]} only if it's not unset. This is to work around
+  # Bash 3 issue with unbound variable.
+  DOCKER=(docker ${DOCKER[@]+"${DOCKER[@]}"})
```

Comment:
Would `:+` be better here? Otherwise will it blow up if its undefined?

https://github.com/kubernetes/kubernetes/blob/5d2b0de95df04e79544f33f4d1f119a1e51defed/hack/ginkgo-e2e.sh#L143

Reply:
According to my reference table, both `+` and `:+` work the same for unset parameters and set non-null parameters. The only difference is if the expanded element is null (there is only one element in the array and it's `""` ). `:+` substitutes a null then, while `+` substitutes the null string. Maybe the null value is better. I'll change this.